cfme/tests/control/test_actions.py

```diff
@@ -31,188 +29,109 @@
-            kwargs = {"flavor_name": provider.data.provisioning.get('instance_type')}
+        if "small_template" in provider.data.templates:
+            kwargs = {"flavor_name": provider.data.provisioning.get("instance_type")}
     elif provider.one_of(SCVMMProvider):
         kwargs = {
-            "host_group": provider.data.get("provisioning", {}).get("host_group", "All Hosts")}
+            "host_group": provider.data.get("provisioning", {}).get("host_group", "All Hosts")
+        }
     else:
         kwargs = {}
 
     collection = provider.appliance.provider_based_collection(provider)
     vm = collection.instantiate(vm_name, provider, template_name)
-
-    try:
-        deploy_template(
-            provider.key,
-            vm_name,
-            template_name=template_name,
-            allow_skip="default",
-            power_on=True,
-            **kwargs
-        )
-    except TimedOutError:
-        message = "{} failed provisioning, check its status!".format(provider.key)
-        logger.exception(message)
-        try:
-            vm.cleanup_on_provider()
-        except TimedOutError:
-            logger.error("Could not delete VM %s!", vm_name)
-    finally:
-        # If this happened, we should skip all tests from this provider in this module
-        pytest.skip(message)
-
+    vm.create_on_provider(delete_on_failure=True, find_in_cfme=True, **kwargs)
     request.addfinalizer(vm.cleanup_on_provider)
-
-    # Make it appear in the provider
     provider.refresh_provider_relationships()
 
-    # Get the REST API object
-    api = wait_for(
-        get_vm_object,
-        func_args=[provider.appliance, vm_name],
-        message="VM object {} appears in CFME".format(vm),
-        fail_condition=None,
-        num_sec=600,
-        delay=15,
-    )[0]
-
-    return VMWrapper(provider, vm_name, api)
+    return vm
 
 
-@pytest.fixture(scope="module")
-def vm(request, provider, setup_provider_modscope, small_template_modscope, vm_name):
-    return _get_vm(request, provider, small_template_modscope.name, vm_name)
+@pytest.fixture
+def vm(request, provider, setup_provider_modscope, small_template_modscope):
+    return _get_vm(request, provider, small_template_modscope.name)
 
 
-@pytest.fixture(scope="module")
+@pytest.fixture
 def vm_big(request, provider, setup_provider_modscope, big_template_modscope):
-    return _get_vm(
-        request,
-        provider,
-        big_template_modscope.name,
-        random_vm_name("action", max_length=16)
-    )
+    return _get_vm(request, provider, big_template_modscope.name)
 
 
 @pytest.fixture(scope="module")
 def policy_name():
     return "action_testing: policy {}".format(fauxfactory.gen_alphanumeric())
 
 
-@pytest.fixture(scope="module")
+@pytest.fixture
 def policy_profile_name():
     return "action_testing: policy profile {}".format(fauxfactory.gen_alphanumeric())
 
 
-@pytest.fixture(scope="module")
+@pytest.fixture
```

Comment:
See comments below, testing now on PRT with function scope.

Reply:
@digitronik See the results from the recent PRT run, 4 out of 15 tests failed when using module scope. The problem is that if a test fails, it is hard to anticipate at which point it will fail and what cleanup will be necessary if you're using the same VM for every test. For that reason, I am going to revert to using the function scope.

Reply:
@digitronik I have since figured out what was causing the issue. The `test_action_prevent_vm_retire` was retiring the VM, and the test was broken so it didn't catch the fact that the retirement was not prevented, so it marked it as a pass. Should be fixed now.